src/minishell.c

```diff
@@ -71,54 +71,53 @@ void	executecmd(char *tokens[], char *envp[])
 	wait(NULL);
 }
 
+static void	looper(char *cmd, char *envp[], t_elst *lst, int debug_mode)
+{
+	t_sent	*sent;
+	t_deque	*deque;
+
+	deque = deque_init();
+	parsecmd(cmd, deque, lst, debug_mode);
+	sent = deque->end;
+
+	if (debug_mode)
+	{
+		ft_printf("\n");
+		sent_print(&sent);
+		ft_printf("\n");
+		ft_printf("------ result ------\n");
+	}
+
+	while (0 < deque->size)
+		executecmd(deque_pop_back(deque)->tokens, envp);
+	sent_delall(&sent);
+	deque_del(deque);
+	return ;
+}
+
 int	main(int argc, char *argv[], char *envp[])
 {
+	int		debug_mode;
 	char	cmd[MAX_COMMAND_LEN];
-	t_deque	*deque;
 	t_elst	*lst;
 
-	(void)envp;
-	if (argc > 1 && argv)
+	debug_mode = FALSE;
+	if (ft_strequ(argv[1], "--debug") || ft_strequ(argv[1], "-d"))
+		debug_mode = TRUE;
+	else if (argc > 1 && argv)
+	{
 		ft_putstr_fd("Invalid arguments. Try ./minishell\n", 2);
+		return (0);
+	}
```

Comment:
Discovered that Minishell was being executed in situations where it should not have been executed due to an error, so modified it so that Minishell would return without executing in that error situation.

---
에러로 인해 미셸이 실행되지 않아야 하는 상황에서 실행되는 것을 발견하여 해당 오류 상황에서 미셸이 실행하지 않고 리턴하도록 수정했습니다.